service/matching/matching_engine.go

```diff
@@ -569,11 +579,70 @@ pollLoop:
 
 			continue pollLoop
 		}
+
+		getHistResp, err := e.getHistoryForPollWorkflowTaskQueue(ctx, namespaceID, task, resp)
```

Comment:
I thought for workflow task case, the history can be returned from recordWorkflowTaskStarted call, without making a separate call to history service.